src/vm/vm.ts

```diff
@@ -240,54 +250,71 @@ function step(n: number = 0): void {
     .getRange(`${resColumn}${n + 2}`)
     .getDisplayValue();
 
-  let newPc: string | number;
   switch (instruction.PcUpdate) {
     case PcUpdates.Jump:
-      newPc = resValue;
+      runSheet
+        .getRange(`${pcColumn}${n + 2 + 1}`)
+        .setFormula(`=${resColumn}${n + 2}`);
       break;
     case PcUpdates.JumpRel:
-      newPc = pc + Number(resValue);
+      runSheet
+        .getRange(`${pcColumn}${n + 2 + 1}`)
+        .setFormula(`=${pcColumn}${n + 2} + ${resValue}`);
       break;
     case PcUpdates.Jnz:
-      newPc = pc + Number(dstValue === 0 ? size(instruction) : op1Value);
+      runSheet
+        .getRange(`${pcColumn}${n + 2 + 1}`)
+        .setFormula(
+          `=${pcColumn}${n + 2} + ${dstValue === 0 ? size(instruction) : op1Value}`,
```

Comment:
to be confirmed but it's possible that in this context op1 is always res because resLogic is always op1

Reply:
In the cairo white paper, I read that in case of jnz, res value is unused but not op1

Reply:
are you mentioning this ?
<img width="373" alt="image" src="https://github.com/ClementWalter/cairo-vm-gs/assets/18620296/aab79a4a-05ae-4762-a960-420050cc5c65">
let's investigate but I can't see why it's useful. Let's ask @spapini

Reply:
@zmalatrax an idea?